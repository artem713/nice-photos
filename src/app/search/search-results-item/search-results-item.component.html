<img #img [src]="item.smallUrl"
     (load)="setLoadingDone()"
     (error)="setLoadingError(); hideImage(img)">

<span *ngIf="isLoadingInProgress" i18n>Loading...</span>
<span *ngIf="isLoadingError" i18n>Photo can't be loaded</span>