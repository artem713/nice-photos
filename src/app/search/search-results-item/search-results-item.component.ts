import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-search-results-item',
  templateUrl: './search-results-item.component.html',
  styles: ['span {font-family: \'Nunito\', serif; color: rgba(0, 0, 0, 0.5);}']
})

export class SearchResultsItemComponent {
  @Input() item; // TODO: add type after search component refactoring

  get isLoadingInProgress() {
    return !this.item.isLoadingDone && !this.item.isLoadingError;
  }

  get isLoadingError() {
    return this.item.isLoadingError;
  }

  setLoadingDone() {
    this.item.isLoadingDone = true;
  }

  setLoadingError() {
    this.item.isLoadingError = true;
  }

  hideImage(img) {
    img.style.display = 'none';
  }
}