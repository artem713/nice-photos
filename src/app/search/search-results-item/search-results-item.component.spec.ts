import { TestBed, ComponentFixture, async } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { SearchResultsItemComponent } from './search-results-item.component';

describe('SearchResultsItemComponent', () => {
  const photoWithValidImage = { smallUrl: 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==' };
  const photoWithInvalidImage = { smallUrl: 'data:image/jpeg;base64,' };
  let component: SearchResultsItemComponent;
  let fixture: ComponentFixture<SearchResultsItemComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [SearchResultsItemComponent]
    });

    fixture = TestBed.createComponent(SearchResultsItemComponent);
    component = fixture.componentInstance;
  });

  it('should be rendered correctly when image is not loaded yet', () => {
    component.item = photoWithValidImage;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  // TODO: find mock for onload event
  xit('should be rendered correctly when image has loaded', async(() => {
    expect.assertions(1);
    component.item = photoWithValidImage;
    fixture.detectChanges();

    const debugElem = fixture.debugElement.query(By.css('img'));
    const elem = debugElem.nativeElement;

    elem.onload = () => {
      fixture.detectChanges();

      expect(fixture).toMatchSnapshot();
    };
  }));

  // TODO: find mock for onerror event
  xit('should be rendered correctly when image has loading error', async(() => {
    expect.assertions(1);
    component.item = photoWithInvalidImage;
    fixture.detectChanges();

    const debugElem = fixture.debugElement.query(By.css('img'));
    const elem = debugElem.nativeElement;

    elem.onerror = () => {
      fixture.detectChanges();

      expect(fixture).toMatchSnapshot();
    };
  }));
});