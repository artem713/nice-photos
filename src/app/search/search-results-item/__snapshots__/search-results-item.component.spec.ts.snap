// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsItemComponent should be rendered correctly when image is not loaded yet 1`] = `
<app-search-results-item
  item={[Function Object]}
>
  <img
    src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
  />
  
  <span>
    Loading...
  </span>
  
</app-search-results-item>
`;
